src/Desktop/ArcGISRuntime.Desktop.Samples/Samples/Security/OAuth/MainWindow.xaml.cs

```diff
@@ -0,0 +1,304 @@
+            }
+
+            // Register this server with AuthenticationManager
+            AuthenticationManager.Current.RegisterServer(serverInfo);
+
+            // Use the OAuthAuthorize class in this project to create a new web view that contains the OAuth challenge handler.
+            AuthenticationManager.Current.OAuthAuthorizeHandler = new OAuthAuthorize();
+
+            // Use a function in this class to challenge for credentials
+            AuthenticationManager.Current.ChallengeHandler = new ChallengeHandler(CreateCredentialAsync);
+        }
+
+        private async void DisplaySecureMap()
+        {
+            // Display a web map hosted in a portal. If the web map item is secured, AuthenticationManager will
+            // challenge for credentials
+            try
+            {
+                // Connect to a portal (ArcGIS Online, for example)
+                ArcGISPortal arcgisPortal = await ArcGISPortal.CreateAsync(new Uri(ServerUrl));
+                // Get a web map portal item using its ID
+                // If the item is secured (not shared publicly) the user will be challenged for credentials at this point
+                ArcGISPortalItem portalItem = await ArcGISPortalItem.CreateAsync(arcgisPortal, WebMapId);
+                // Create a new map with the portal item
+                Map myMap = new Map(portalItem);
+
+                // Assign the map to the MapView.Map property to display it in the app
+                MyMapView.Map = myMap;
+                await myMap.RetryLoadAsync();
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("Error displaying map: " + ex.Message);
+            }
+        }
+
+        public async Task<Credential> CreateCredentialAsync(CredentialRequestInfo info)
+        {
+            // ChallengeHandler function for AuthenticationManager that will be called whenever access to a secured
+            // resource is attempted
+            OAuthTokenCredential credential = null;
+
+            try
+            {
+                // Create generate token options if necessary
+                if (info.GenerateTokenOptions == null)
+                {
+                    info.GenerateTokenOptions = new GenerateTokenOptions();
+                }
+
+                // AuthenticationManager will handle challenging the user for credentials
+                credential = await AuthenticationManager.Current.GenerateCredentialAsync
+                    (
+                            info.ServiceUri,
+                            info.GenerateTokenOptions
+                    ) as OAuthTokenCredential;
+            }
+            catch (Exception ex)
+            {
+                // Exception will be reported in calling function
+                throw (ex);
+            }
+
+            return credential;
+        }
+    }
+
+    public class OAuthAuthorize : IOAuthAuthorizeHandler
```

Comment:
Add comments what this class does. Maybe also add why this is needed in desktop and not in uwp?